### PR TITLE
convert group navigation to scope / store

### PR DESCRIPTION
**Pull request by bluwave**
[ignore-commit-lint]

### CODE REVIEW

**Comment by inst-danger** [comment]
Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 92.75% | 92.86% | -0.11%
**Student** | 94.55% | 94.51% | 0.03%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.54% | 92.54% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Student/Student/Groups/GroupNavigation/GroupNavigationPresenter.swift [hunk 1]
```diff
@@ -19,55 +19,47 @@ import Core
...
 
-        if let u = u { useCase = u } else { useCase = GroupNavigationUseCase(context: context, env: env) }
-
-        frc = env.subscribe(Tab.self, .context(context))
-        groupFrc = env.subscribe(Group.self, .details(groupID))
-        frc.delegate = self
-        groupFrc.delegate = self
+    func viewIsReady() {
+        groups.refresh()
+        tabs.refresh()
```
Comment (vanwagonet) [inline Student/Student/Groups/GroupNavigation/GroupNavigationPresenter.swift]: Do we want to `update()` here, so nav color could get populated early?

Comment (bluwave) [reply Student/Student/Groups/GroupNavigation/GroupNavigationPresenter.swift]: yes, good suggestion, thanks

Comment (bluwave) [reply Student/Student/Groups/GroupNavigation/GroupNavigationPresenter.swift]: fixed

File: Student/StudentUnitTests/Groups/GroupNavigation/GroupNavigationPresenterTests.swift [hunk 2]
```diff
@@ -87,10 +85,12 @@ class GroupNavigationPresenterTests: PersistenceTestCase {
 extension GroupNavigationPresenterTests: GroupNavigationViewProtocol {
     func updateNavBar(title: String, backgroundColor: UIColor) {
         resultingTitle = title
+        expectation.fulfill()
+        print("tabs: \(presenter.tabs.count)")
+        print("first: \(String(describing: presenter.tabs.first?.id))")
```
Comment (armstrongnate) [inline Student/StudentUnitTests/Groups/GroupNavigation/GroupNavigationPresenterTests.swift]: Don't want prints.

**Pull request merged**